ICSharpCode.Decompiler/IL/ILReader.cs

```diff
@@ -996,7 +996,8 @@ ILInstruction Push(ILInstruction inst)
 			Debug.Assert(inst.ResultType != StackType.Void);
 			IType type = compilation.FindType(inst.ResultType.ToKnownTypeCode());
 			var v = new ILVariable(VariableKind.StackSlot, type, inst.ResultType, inst.ILRange.Start);
-			v.Name = "S_" + inst.ILRange.Start.ToString("x4");
+			var start = inst.ILRange.Start;
+			v.Name = start == 0 ? "S_0000" : "S_" + start.ToString("x4");
```

Comment:
Oh interesting, those are a lot generated strings. And (almost?) all of them will never be used (except for debugging output).
Is the `start == 0` special case all that helpful? I would expect the values to be evenly distributed across the IL body.
We should try to see if we can get away with `v.Name == null` here and generate this default name only on-demand (when writing debug output).

Reply:
Of course the fact that we generate an ILVariable object at all is already suboptimal -- if we combined the ILReader with a first (simple) round of inlining, we could probably avoid allocating at least half of these variables.

Reply:
I'm not sure why but when I tested this then `ILRange.Start` was always 0. Maybe there is something wrong elsewhere.
I agree that a lot of these ILVariables are unnecessary and a more general solution (like inlining or using null) works better here.

I started with some optimization work mainly to get familiar with the codebase again.

Reply:
Oh right, the `ILRange` only gets set in `ReadInstructions()`, which is after the `Push()` call.
The variable name we see in debug output is that assigned by the `CollectStackVariablesVisitor`, after the variables from different control flow paths are unified.
I guess that means the initial name can safely stay `null`, as there's no way to observe it outside of the `ILReader`.